### ISSUE TITLE
drawerOpened may be false when opening the drawer by sliding

### ISSUE THREAD

**Issue by yoavrofe** [flutter/flutter#27674]
When checking on `onWillPop`, `scaffoldKey.currentState.isDrawerOpen` may report `false` if the drawer was opened by sliding the finger from the edge of the screen.

**Comment by zoechi** [comment]
Please add the output of `flutter doctor -v`.

To investigate we would need a minimal runnable reproduction
as a single file so that we can just copy your code into `lib/main.dart` of a new project and run to reproduce.

**Comment by yoavrofe** [comment]
I tried it, and could reproduce it only once on a simple application (tried to make it a bit more complex with stageful widgets). On my larger app it happens more frequently.

[main.dart.txt](https://github.com/flutter/flutter/files/2844827/main.dart.txt)

```
flutter doctor -v
[✓] Flutter (Channel master, v1.2.1-pre.125, on Mac OS X 10.14.2 18C54, locale en-IL)
    • Flutter version 1.2.1-pre.125 at /Users/yoavrofe/flutter
    • Framework revision 4e9cfa8f85 (10 hours ago), 2019-02-07 20:40:28 -0500
    • Engine revision b0370c13c3
    • Dart version 2.1.2 (build 2.1.2-dev.0.0 6c680a0ebd)

[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/yoavrofe/Library/Android/sdk
    • Android NDK at /Users/yoavrofe/Library/Android/sdk/ndk-bundle
    • Platform android-28, build-tools 28.0.3
    • ANDROID_HOME = /Users/yoavrofe/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)
    • All Android licenses accepted.

[✓] iOS toolchain - develop for iOS devices (Xcode 10.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.1, Build version 10B61
    • ios-deploy 1.9.4
    • CocoaPods version 1.5.2

[✓] Android Studio (version 3.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 31.3.3
    • Dart plugin version 182.5124
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)

[✓] VS Code (version 1.31.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 2.22.3

[✓] Connected device (1 available)
    • POCOPHONE F1 • df1855b5 • android-arm64 • Android 9 (API 28)

• No issues found!
```

**Comment by zoechi** [comment]
I was not yet able to reproduce 
```
[✓] Flutter (Channel master, v1.2.1-pre.117, on Mac OS X 10.14.3 18D42, locale en-AT)
```

Please let us know if you find anything else that might make the reproduction more reliable.

**Comment by fellow7000** [comment]
@zoechi same stuff here - just have a Scaffold with drawer and isDrawerOpen is false 3-4 time out of each 10 ones if I open it by sliding. The widget is relative complex. 

This does not happen if Drawer is opened by icon or programatically, only by slider.

Flutter channel master, v1.6.8-pre.43

**Comment by evaldobratti** [comment]
Same problem here

[gist](https://gist.github.com/evaldobratti/a6acdd64ea450beac19befde51493e85) and [video](https://youtu.be/6oJp2zxYyqA)

if you open the drawer with a sudden but small movement, you are able to reproduce it, even in the emulator

**Comment by evaldobratti** [comment]
The problem seems to be that the animation keeps running, opening the drawer fully, without breaking the `0.5` treshold in `DrawerControllerState._move`, the callbacks are not called and the view is not compatible with the state.

**Comment by lsi8** [comment]
Please fix this!

**Comment by honey-system** [comment]
Same problem

**Comment by moodstubos** [comment]
unfortunately it is still a problem (dev channel 1.10.14).

as @fellow7000 wrote:
> just have a Scaffold with drawer and isDrawerOpen is false 3-4 time out of each 10 ones if I open it by sliding. The widget is relative complex.

i check isDrawerOpen in onWillPop

**Comment by AliShareei** [comment]
Same Problem in onWillPop event ... !

**Comment by moodstubos** [comment]
@AliShareei please also use the thumb-up button on the very first post.

**Comment by TahaTesser** [comment]
Looks like it returns true even when drawer was opened by sliding the finger from the edge of the screen.

<details> 
<summary>Code Sample</summary> 

``` 
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatefulWidget {
  @override
  _MyAppState createState() {
    return _MyAppState();
  }
}

class _MyAppState extends State<MyApp> {
  final scaffoldKey = GlobalKey<ScaffoldState>();

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: Scaffold(
        key: scaffoldKey,
        drawer: new DRW(),
        appBar: AppBar(
          title: Text("Drawer Bug!"),
        ),
        body: WillPopScope(
          onWillPop: () {
            print("Drawer is open: ${scaffoldKey.currentState.isDrawerOpen}");
          },
          child: Center(
            child: Text('Hello World!'),
          ),
        ),
      ),
    );
  }
}

class DRW extends StatefulWidget {
  const DRW({
    Key key,
  }) : super(key: key);

  @override
  _DRWState createState() {
    return _DRWState();
  }
}

class _DRWState extends State<DRW> {
  @override
  Widget build(BuildContext context) {
    return Container(
      color: Colors.red,
      width: 100,
      child: Column(
        children: <Widget>[
          Icon(Icons.access_time),
        ],
      ),
    );
  }
}

``` 

</details> 

```
I/flutter (30251): Drawer is open: true
```

<details> 
<summary>flutter doctor -v</summary> 

``` 
[✓] Flutter (Channel dev, 1.19.0-1.0.pre, on Mac OS X 10.15.5 19F96, locale
    en-GB)
    • Flutter version 1.19.0-1.0.pre at /Users/tahatesser/Code/flutter_dev
    • Framework revision 456d80b9dd (3 weeks ago), 2020-05-11 11:45:03 -0400
    • Engine revision d96f962ca2
    • Dart version 2.9.0 (build 2.9.0-7.0.dev 092ed38a87)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /Users/tahatesser/Code/sdk
    • Platform android-29, build-tools 29.0.3
    • ANDROID_HOME = /Users/tahatesser/Code/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.5)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.5, Build version 11E608c
    • CocoaPods version 1.9.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 45.1.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.45.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.10.2

[✓] Connected device (5 available)
    • SM M305F   • 32003c30dc19668f                     • android-arm64  •
      Android 10 (API 29)
    • iPhone 11  • 0EC80516-8F99-4A0C-89ED-2273579862F9 • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-13-5 (simulator)
    • macOS      • macOS                                • darwin-x64     • Mac
      OS X 10.15.5 19F96
    • Web Server • web-server                           • web-javascript •
      Flutter Tools
    • Chrome     • chrome                               • web-javascript •
      Google Chrome 83.0.4103.61

• No issues found!
``` 

</details> 


Could everyone who still has this problem please file a new issue with the exact description of what happens, logs and the output of `flutter doctor -v`.
All system setups can be slightly different, so it's always better to open new issues and reference related issues.

**Comment by evaldobratti** [comment]
I just tested with version `1.16.3-dev`, last one available on `asdf` and its still happening.

@TahaTesser have you checked the video I posted [here](https://github.com/flutter/flutter/issues/27674#issuecomment-505027798)?

I debuged this issue the time I had this problem and kinda found the problem [here](https://github.com/flutter/flutter/issues/27674#issuecomment-505473372)

It words indeed, but with a fast and little movement, it doesn't work all times.

**Comment by TahaTesser** [comment]
hI @evaldobratti 
Issue isn't reproducible on `Channel dev, 1.19.0-1.0.pre` 
Can you please try it on latest `dev`?
Thank you

**Comment by evaldobratti** [comment]
Hi @TahaTesser 

I was able to reproduce on the version you mentioned

```
[✓] Flutter (Channel dev, 1.19.0-1.0.pre, on Linux, locale pt_BR.UTF-8)
    • Flutter version 1.19.0-1.0.pre at /home/ebratti/.asdf/installs/flutter/.19.0-1.0.pre-dev
    • Framework revision 456d80b9dd (3 weeks ago), 2020-05-11 11:45:03 -0400
    • Engine revision d96f962ca2
    • Dart version 2.9.0 (build 2.9.0-7.0.dev 092ed38a87)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /opt/Android/Sdk
    • Platform android-29, build-tools 29.0.3
    • ANDROID_HOME = /opt/Android/Sdk
    • Java binary at: /opt/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Android Studio (version 3.5)
    • Android Studio at /opt/android-studio
    • Flutter plugin version 44.0.1
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[✓] VS Code (version 1.45.1)
    • VS Code at /usr/share/code
    • Flutter extension version 3.11.0

[✓] Connected device (1 available)
    • moto g 6 plus • 0046510260 • android-arm64 • Android 9 (API 28)

• No issues found!
```

I recorded another video https://youtu.be/7PJnvbSLa30

Take a look at the length of the movement needed to open the drawer. If you make the movement fast and short from the edge of the screen, you'll reproduce it

**Comment by TahaTesser** [comment]
@evaldobratti 
I tried for fifth time and still not able to reproduce this issue 
but i will reopen it since you can confirm, could device specific

**Comment by evaldobratti** [comment]
It's hard to reproduce at first, but once you do it, you can reproduce it everytime.
The key steps to reproduce it are
1. Start the movement from the edge
2. Do it quickly enough to open the drawer, but not too long

I recorded it again pressing in the appbar to show touching better, take a look @TahaTesser 
https://www.youtube.com/watch?v=1hvd0F7pkRI

**Comment by chunhtai** [comment]
@evaldobratti 
I still cannot reproduce it, which device are you using to reproduce the error? Can you also try on different devices and see if you can reproduce it?

**Comment by evaldobratti** [comment]
You can reproduce it using an emulator and clicks as well, in this one I'm running a nexus 4 api 26 768x1280 xhdpi

Pay attention at the label in the body screen, cus some times it works and sometimes not

https://youtu.be/N_uGmiROKSg

It's really harder than in an actual touch device though

**Comment by evaldobratti** [comment]
Remember to release your touching as soon as possible, it should start from the edge going to the center of the screen, but with the smallest traveled distance possible, the acceleration should be still enough to open the drawer, then the opened flag will be false even though the drawer is fully open.

**Comment by AliShareei** [comment]
I fix my problem on onWillPop event (I use this event for double tap exit)
Because of this bug, I had to check canPop(), anyway it solved my problem.
 `Navigator.of(context).canPop()`

**Comment by TahaTesser** [comment]
@yoavrofe  @evaldobratti 

Tried to reproduce again, no luck 


<details>
<summary>code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatefulWidget {
  @override
  _MyAppState createState() {
    return _MyAppState();
  }
}

class _MyAppState extends State<MyApp> {
  final scaffoldKey = GlobalKey<ScaffoldState>();

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: Scaffold(
        key: scaffoldKey,
        drawer: new DRW(),
        appBar: AppBar(
          title: Text("Drawer Bug!"),
        ),
        body: WillPopScope(
          onWillPop: () {
            print("Drawer is open: ${scaffoldKey.currentState.isDrawerOpen}");
          },
          child: Center(
            child: Text('Hello World!'),
          ),
        ),
      ),
    );
  }
}

class DRW extends StatefulWidget {
  const DRW({
    Key key,
  }) : super(key: key);

  @override
  _DRWState createState() {
    return _DRWState();
  }
}

class _DRWState extends State<DRW> {
  @override
  Widget build(BuildContext context) {
    return Container(
      color: Colors.red,
      width: 100,
      child: Column(
        children: <Widget>[
          Icon(Icons.access_time),
        ],
      ),
    );
  }
}
```
</details>


<details>
<summary>logs</summary>

```bash
[        ] Compiling dart to kernel with 0 updated files
[        ] <- recompile package:triage/main.dart dfb23a39-77e6-4836-9486-201249e7c040
[        ] <- dfb23a39-77e6-4836-9486-201249e7c040
[  +60 ms] Updating files
[ +543 ms] DevFS: Sync finished
[        ] Syncing files to device Android SDK built for x86... (completed in 608ms)
[        ] Synced 1.8MB.
[   +1 ms] <- accept
[   +3 ms] Connected to _flutterView/0xdce32c10.
[   +1 ms] Flutter run key commands.
[   +1 ms] r Hot reload. 🔥🔥🔥
[        ] R Hot restart.
[        ] h Repeat this help message.
[        ] d Detach (terminate "flutter run" but leave application running).
[        ] c Clear the screen
[        ] q Quit (terminate the application on the device).
[        ] An Observatory debugger and profiler on Android SDK built for x86 is available at: http://127.0.0.1:60096/Wo_vKYP5Ezs=/
[ +234 ms] D/EGL_emulation(10990): eglMakeCurrent: 0xe021a0c0: ver 3 0 (tinfo 0xe025b030)
[   +8 ms] D/eglCodecCommon(10990): setVertexArrayObject: set vao to 0 (0) 1 0
[  +52 ms] I/OpenGLRenderer(10990): Davey! duration=1602ms; Flags=1, IntendedVsync=928588158430, Vsync=928588158430, OldestInputEvent=9223372036854775807, NewestInputEvent=0, HandleInputStart=928590219904, AnimationStart=928590281904, PerformTraversalsStart=928590340904, DrawStart=930122613904, SyncQueued=930133139904, SyncStart=930134746904, IssueDrawCommandsStart=930137811904, SwapBuffers=930169921904, FrameCompleted=930192671904, DequeueBufferDuration=12790000, QueueBufferDuration=883000, 
[   +6 ms] I/Choreographer(10990): Skipped 95 frames!  The application may be doing too much work on its main thread.
[ +280 ms] D/EGL_emulation(10990): eglMakeCurrent: 0xf347f340: ver 3 0 (tinfo 0xe02ccfc0)
[   +4 ms] D/eglCodecCommon(10990): setVertexArrayObject: set vao to 0 (0) 1 0
[+49953 ms] I/flutter (10990): Drawer is open: true
[+1027 ms] I/flutter (10990): Drawer is open: true
[ +826 ms] I/flutter (10990): Drawer is open: true
[ +149 ms] I/flutter (10990): Drawer is open: true
[ +176 ms] I/flutter (10990): Drawer is open: true
[ +169 ms] I/flutter (10990): Drawer is open: true
[+16128 ms] I/flutter (10990): Drawer is open: true
[ +766 ms] I/flutter (10990): Drawer is open: true
[ +142 ms] I/flutter (10990): Drawer is open: true
[ +146 ms] I/flutter (10990): Drawer is open: true
[ +169 ms] I/flutter (10990): Drawer is open: true
[ +147 ms] I/flutter (10990): Drawer is open: true
[+13041 ms] I/flutter (10990): Drawer is open: true
[ +544 ms] I/flutter (10990): Drawer is open: true
[ +142 ms] I/flutter (10990): Drawer is open: true
[ +143 ms] I/flutter (10990): Drawer is open: true
[+51284 ms] I/flutter (10990): Drawer is open: true
[+7306 ms] I/flutter (10990): Drawer is open: true
[+3443 ms] I/flutter (10990): Drawer is open: true
[ +668 ms] I/flutter (10990): Drawer is open: true
[ +174 ms] I/flutter (10990): Drawer is open: true
[ +156 ms] I/flutter (10990): Drawer is open: true
[ +161 ms] I/flutter (10990): Drawer is open: true
[+114376 ms] I/flutter (10990): Drawer is open: true
[ +728 ms] I/flutter (10990): Drawer is open: true
[ +157 ms] I/flutter (10990): Drawer is open: true
[ +154 ms] I/flutter (10990): Drawer is open: true
[ +153 ms] I/flutter (10990): Drawer is open: true
[ +145 ms] I/flutter (10990): Drawer is open: true
[ +159 ms] I/flutter (10990): Drawer is open: true
```
</details>

<details>
<summary>flutter doctor -v</summary>

```bash
[✓] Flutter (Channel stable, 1.22.5, on macOS 11.1 20C69 darwin-x64, locale en-GB)
    • Flutter version 1.22.5 at /Users/tahatesser/Code/flutter_stable
    • Framework revision 7891006299 (6 weeks ago), 2020-12-10 11:54:40 -0800
    • Engine revision ae90085a84
    • Dart version 2.10.4

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Volumes/Extreme/SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Volumes/Extreme/SDK
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Volumes/Extreme/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    • CocoaPods version 1.10.1

[!] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.52.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.18.1

[✓] Connected device (2 available)
    • RMX2001 (mobile)     • EUYTFEUSQSRGDA6D          • android-arm64 • Android 10 (API 29)
    • Taha’s iPad (mobile) • 00008020-000255113EE8402E • ios           • iOS 14.3

! Doctor found issues in 1 category.
```
```bash
[✓] Flutter (Channel master, 1.26.0-2.0.pre.419, on macOS 11.1 20C69 darwin-x64, locale en-GB)
    • Flutter version 1.26.0-2.0.pre.419 at /Users/tahatesser/Code/flutter_master
    • Framework revision 913d5933c1 (4 hours ago), 2021-01-19 22:34:03 -0800
    • Engine revision b58dbc88fc
    • Dart version 2.12.0 (build 2.12.0-240.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Volumes/Extreme/SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Volumes/Extreme/SDK
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Volumes/Extreme/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.52.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.18.1

[✓] Connected device (4 available)
    • RMX2001 (mobile)     • EUYTFEUSQSRGDA6D          • android-arm64  • Android 10 (API 29)
    • Taha’s iPad (mobile) • 00008020-000255113EE8402E • ios            • iOS 14.3
    • macOS (desktop)      • macos                     • darwin-x64     • macOS 11.1 20C69 darwin-x64
    • Chrome (web)         • chrome                    • web-javascript • Google Chrome 87.0.4280.141

• No issues found!
```

</details>

**Comment by pedromassangocode** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. 
We are therefore reluctantly going to close this bug for now. 
Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.

Could everyone who still has this problem please file a new issue with the exact description of what happens, logs, and the output of `flutter doctor -v`.
All system setups can be slightly different, so it's always better to open new issues and reference related issues.
